odd-platform-api/src/main/java/com/provectus/oddplatform/repository/DatasetFieldRepositoryImpl.java

```diff
@@ -166,4 +176,85 @@ public void updateSearchVectors(final long datasetFieldId) {
             Map.of(labelName, LABEL.NAME)
         ).execute();
     }
+
+    @Override
+    @Transactional
+    public DatasetFieldDto updateDatasetField(final long datasetFieldId,
+                                              final DatasetFieldUpdateFormData datasetFieldUpdateFormData) {
+        setDescription(datasetFieldId, datasetFieldUpdateFormData.getDescription());
+
+        final Set<String> names = new HashSet<>(datasetFieldUpdateFormData.getLabelNames());
+
+        final List<LabelPojo> currentLabels = labelRepository.listByDatasetFieldId(datasetFieldId);
+
+        final List<LabelPojo> existingLabels = labelRepository.listByNames(names);
+
+        final List<String> existingLabelsNames = existingLabels.stream()
+            .map(LabelPojo::getName)
+            .collect(Collectors.toList());
+
+        final Set<String> labelNames = currentLabels.stream()
+            .map(LabelPojo::getName)
+            .collect(Collectors.toSet());
+
+        final List<Long> idsToDelete = currentLabels.stream()
+            .filter(l -> !names.contains(l.getName()))
+            .map(LabelPojo::getId)
+            .collect(Collectors.toList());
+
+        labelRepository.deleteRelations(datasetFieldId, idsToDelete);
+
+        final List<LabelPojo> labelsToCreate = names.stream()
+            .filter(n -> !labelNames.contains(n) && !existingLabelsNames.contains(n))
+            .map(n -> new LabelPojo().setName(n))
+            .collect(Collectors.toList());
+
+        final List<Long> createdIds = labelRepository.bulkCreate(labelsToCreate).stream()
+            .map(LabelPojo::getId)
+            .collect(Collectors.toList());
+
+        final Set<Long> toRelate = Stream.concat(
+            createdIds.stream(),
+            existingLabels.stream().map(LabelPojo::getId).filter(not(idsToDelete::contains))
+        ).collect(Collectors.toSet());
+
+        labelRepository.createRelations(datasetFieldId, toRelate);
+
+        final Set<LabelPojo> labelPojos = Stream.concat(
+                labelsToCreate.stream(),
+                existingLabels.stream())
+            .collect(Collectors.toSet());
+
+        final DatasetFieldDto dto = getDto(datasetFieldId);
+        dto.setLabelPojos(labelPojos);
+
+        updateSearchVectors(datasetFieldId);
```

Comment:
updateSearchVectors is called inside setDescription as well, so it's possible that it will be executed twice